dicom_util/src/main/java/com/google/cloud/healthcare/DicomWebClient.java

```diff
@@ -78,17 +80,26 @@ public JSONArray qidoRs(String path) throws IDicomWebClient.DicomWebException {
           requestFactory.buildGetRequest(new GenericUrl(serviceUrlPrefix + "/" + path));
       HttpResponse httpResponse = httpRequest.execute();
 
-      if (!httpResponse.isSuccessStatusCode()) {
-        throw new IDicomWebClient.DicomWebException(
-            String.format(
-                "QidoRs: %d, %s", httpResponse.getStatusCode(), httpResponse.getStatusMessage()));
-      }
       // dcm4che server can return 204 responses.
       if (httpResponse.getStatusCode() == HttpStatusCodes.STATUS_CODE_NO_CONTENT) {
         return new JSONArray();
       }
       return new JSONArray(
           CharStreams.toString(new InputStreamReader(httpResponse.getContent(), "UTF-8")));
+    } catch (HttpResponseException e) {
```

Comment:
It would be nice to have this mapping logic in a function which for all three methods (wado, qido, and stow) use. Then we only have to worry about this mapping in one place, rather than having to make changes throughout the codebase when we want to update it. So all three of the methods would call into a function that maps the HTTP status to DIMSE status. And then we can migrate the CSTORE sender to use the DicomWebClient. What do you think?

Reply:
Yes, makes sense.